src/components/file-browser/FileBrowserDirectories.ts

```diff
@@ -0,0 +1,172 @@
+import { Fragment, ReactNode, useState } from 'react'
+import { details, div, h, li, p, span, summary, ul } from 'react-hyperscript-helpers'
+import { Link } from 'src/components/common'
+import { useDirectoriesInDirectory } from 'src/components/file-browser/file-browser-hooks'
+import { basename } from 'src/components/file-browser/file-browser-utils'
+import { icon } from 'src/components/icons'
+import FileBrowserProvider from 'src/libs/ajax/file-browser-providers/FileBrowserProvider'
+import colors from 'src/libs/colors'
+
+
+interface DirectoryContentsProps {
+  provider: FileBrowserProvider
+  isHighlighted: (path: string) => boolean
+  level: number
+  path: string
+  onSelectDirectory: (path: string) => void
+}
+
+const renderDirectoryStatus = (level: number, children: ReactNode) => p({ style: { padding: '0.25rem 0.5rem', margin: `0 0 0.25rem ${level * 1.25}rem` } }, [children])
+
+const DirectoryContents = (props: DirectoryContentsProps) => {
+  const { provider, isHighlighted, level, path, onSelectDirectory } = props
+
+  const {
+    state: { status, directories },
+    hasNextPage,
+    loadNextPage
+  } = useDirectoriesInDirectory(provider, path)
+
+  if (status === 'Error') {
+    return renderDirectoryStatus(level + 1, 'Error loading contents')
+  }
+
+  return h(Fragment, [
+    directories.length > 0 && ul({
+      style: {
+        padding: 0,
+        margin: 0,
+        listStyleType: 'none'
+      }
+    }, [
+      directories.map(directory => li({
+        key: directory.path
+      }, [
+        h(Directory, {
+          provider,
+          isHighlighted,
+          level: level + 1,
+          path: directory.path,
+          onSelectDirectory
+        })
+      ]))
+    ]),
+    status === 'Loading' && renderDirectoryStatus(level, 'Loading...'),
+    hasNextPage && h(Link, {
```

Comment:
Note that Links will become Buttons if they don't have an href defined.

Reply:
I know, but Link seems to be our component for a text only button.

Reply:
Clickable doesn't work?

Reply:
That would lose the visual styles on Link.
https://github.com/DataBiosphere/terra-ui/blob/c31123e6328f56789203ba3a8f31e60786e61446/src/components/common.js#L95-L100

Reply:
Thanks, I now think I have to go back and fix something. :)